manifests/forwarder.pp

```diff
@@ -1,153 +1,226 @@
-  Boolean $purge_outputs                = false,
-  Boolean $purge_props                  = false,
-  Boolean $purge_transforms             = false,
-  Boolean $purge_web                    = false,
-  Optional[String] $pkg_provider        = $splunk::params::pkg_provider,
-  String $forwarder_confdir             = $splunk::params::forwarder_confdir,
-  Hash $forwarder_output                = $splunk::params::forwarder_output,
-  Hash $forwarder_input                 = $splunk::params::forwarder_input,
-  Boolean $create_password              = $splunk::params::create_password,
-  Hash $addons                          = {},
-) inherits splunk::params {
-
-  $virtual_service = $splunk::params::forwarder_service
-  $staging_subdir  = $splunk::params::staging_subdir
-
-  $path_delimiter  = $splunk::params::path_delimiter
-
-  $_package_source = $manage_package_source ? {
-    true  => $splunk::params::forwarder_pkg_src,
-    false => $package_source,
-  }
-
-  #no need for staging the source if we have yum or apt
-  if $pkg_provider != undef and $pkg_provider != 'yum' and $pkg_provider != 'apt' and $pkg_provider != 'chocolatey' {
-    include ::archive::staging
-
-    $src_pkg_filename = basename($_package_source)
-    $pkg_path_parts   = [$archive::path, $staging_subdir, $src_pkg_filename]
-    $staged_package   = join($pkg_path_parts, $path_delimiter)
-
-    archive { $staged_package:
-      source  => $_package_source,
-      extract => false,
-      before  => Package[$package_name],
-    }
-  } else {
-    $staged_package = undef
-  }
-
-  Package  {
-    source          => $pkg_provider ? {
-      'chocolatey' => undef,
-      default      => $manage_package_source ? {
-        true  => pick($staged_package, $_package_source),
-        false => $_package_source,
-      }
-    },
-  }
-
-  package { $package_name:
-    ensure          => $package_ensure,
-    provider        => $pkg_provider,
-    before          => Service[$virtual_service],
-    install_options => $install_options,
-    tag             => 'splunk_forwarder',
-  }
-
-  # Declare addons
-  create_resources('splunk::addon', $addons)
-
-  # Ensure that the service restarts upon changes to addons
-  Package[$package_name] -> Splunk::Addon <||> ~> Service[$virtual_service]
-
-  # Declare inputs and outputs specific to the forwarder profile
-  $tag_resources = { tag => 'splunk_forwarder' }
-  if $forwarder_input {
-    create_resources( 'splunkforwarder_input',$forwarder_input, $tag_resources)
-  }
-  if $forwarder_output {
-    create_resources( 'splunkforwarder_output',$forwarder_output, $tag_resources)
-  }
-  # this is default
-  splunkforwarder_web { 'forwarder_splunkd_port':
-    section => 'settings',
-    setting => 'mgmtHostPort',
-    value   => "${splunkd_listen}:${splunkd_port}",
-    tag     => 'splunk_forwarder',
-  }
-
-  # If the purge parameters have been set, remove all unmanaged entries from
-  # the respective config files.
+# Class splunk::forwarder
+# 
```

Comment:
nit.  Trailing whitespace on every other line.

Reply:
Not sure why puppet-lint doesn't complain.

Reply:
Maybe comments are linted differently?